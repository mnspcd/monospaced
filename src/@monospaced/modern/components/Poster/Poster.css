.mdrn .Poster {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background-color: var(--color-blue-6);
  background-image: url(../../assets/images/desk-x1.png);
  background-position: bottom;
  background-size: cover;
  color: var(--color-white);
}

@media (min-resolution: 2dppx) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-x2.png);
  }
}

@media (min-width: 338px) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-338-x1.png);
  }
}

@media (min-width: 338px) and (min-resolution: 2dppx) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-338-x2.png);
  }
}

@media (min-width: 495px) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-495-x1.png);
  }
}

@media (min-width: 495px) and (min-resolution: 2dppx) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-495-x2.png);
  }
}

@media (min-width: 640px) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-640-x1.png);
  }
}

@media (min-width: 640px) and (min-resolution: 2dppx) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-640-x2.png);
  }
}

@media (min-width: 768px) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-768-x1.png);
  }
}

@media (min-width: 768px) and (min-resolution: 2dppx) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-768-x2.png);
  }
}

@media (min-width: 1024px) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-1024-x1.png);
  }
}

@media (min-width: 1024px) and (min-resolution: 2dppx) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-1024-x2.png);
  }
}

@media (min-width: 1280px) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-1280-x1.png);
  }
}

@media (min-width: 1280px) and (min-resolution: 2dppx) {
  .mdrn .Poster {
    background-image: url(../../assets/images/desk-1280-x2.png);
  }
}

.mdrn .Poster a {
  color: var(--color-white);
}

.mdrn .Poster a:hover {
  color: var(--color-blue-2);
}

.mdrn .Poster a:active {
  color: var(--color-blue-3);
}
