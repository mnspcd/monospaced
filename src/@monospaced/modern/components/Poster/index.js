export { default, PosterContext } from "./Poster";
