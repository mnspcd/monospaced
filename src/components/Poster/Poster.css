.Poster {
  background-color: var(--mnspcd-color-blue-6);
  background-image: url(../../assets/images/desk-x1.png);
  background-position: bottom;
  background-size: cover;
  color: var(--mnspcd-color-white);
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

@media (min-resolution: 2dppx) {
  .Poster {
    background-image: url(../../assets/images/desk-x2.png);
  }
}

@media (min-width: 338px) {
  .Poster {
    background-image: url(../../assets/images/desk-338-x1.png);
  }
}

@media (min-width: 338px) and (min-resolution: 2dppx) {
  .Poster {
    background-image: url(../../assets/images/desk-338-x2.png);
  }
}

@media (min-width: 495px) {
  .Poster {
    background-image: url(../../assets/images/desk-495-x1.png);
  }
}

@media (min-width: 495px) and (min-resolution: 2dppx) {
  .Poster {
    background-image: url(../../assets/images/desk-495-x2.png);
  }
}

@media (min-width: 640px) {
  .Poster {
    background-image: url(../../assets/images/desk-640-x1.png);
  }
}

@media (min-width: 640px) and (min-resolution: 2dppx) {
  .Poster {
    background-image: url(../../assets/images/desk-640-x2.png);
  }
}

@media (min-width: 768px) {
  .Poster {
    background-image: url(../../assets/images/desk-768-x1.png);
  }
}

@media (min-width: 768px) and (min-resolution: 2dppx) {
  .Poster {
    background-image: url(../../assets/images/desk-768-x2.png);
  }
}

@media (min-width: 1024px) {
  .Poster {
    background-image: url(../../assets/images/desk-1024-x1.png);
  }
}

@media (min-width: 1024px) and (min-resolution: 2dppx) {
  .Poster {
    background-image: url(../../assets/images/desk-1024-x2.png);
  }
}

@media (min-width: 1280px) {
  .Poster {
    background-image: url(../../assets/images/desk-1280-x1.png);
  }
}

@media (min-width: 1280px) and (min-resolution: 2dppx) {
  .Poster {
    background-image: url(../../assets/images/desk-1280-x2.png);
  }
}

.Poster a {
  color: var(--mnspcd-color-white);
}

.Poster a:hover {
  color: var(--mnspcd-color-blue-2);
}

.Poster a:active {
  color: var(--mnspcd-color-blue-3);
}
