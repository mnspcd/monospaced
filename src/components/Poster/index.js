export { default, PosterConsumer } from "./Poster";
