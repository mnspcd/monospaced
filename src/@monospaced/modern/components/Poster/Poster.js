import PropTypes from "prop-types";
import React from "react";

import "./Poster.css";

const { Consumer, Provider } = React.createContext();

/**
 * Use `Poster` to display content on an image.
 */
const Poster = ({ children = null, padding = null }) => (
  <Provider value={true}>
    <div className="Poster" style={{ padding }}>
      {children}
    </div>
  </Provider>
);

Poster.propTypes = {
  /**
   * Content to display on the Poster.
   */
  children: PropTypes.node,

  /**
   * A valid CSS padding value for the Poster.
   */
  padding: PropTypes.string,
};

export default Poster;
export { Consumer as PosterConsumer };
