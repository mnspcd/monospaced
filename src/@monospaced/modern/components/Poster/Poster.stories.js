import { text } from "@storybook/addon-knobs";
import React from "react";

import Poster from "./Poster";

export default {
  component: Poster,
  title: "Components/Poster",
};

const ExampleContent = () => (
  <div className="u-monospaced" style={{ padding: "1em" }}>
    Text <br />
    <a href="#">Link</a>
  </div>
);

export const basic = () => <Poster padding="9em 1em"></Poster>;

export const textContent = () => (
  <Poster padding="1.5em 1em">
    <ExampleContent />
  </Poster>
);

export const knobs = () => (
  <Poster padding={text("padding", "1.5em 1em")}>
    <div className="u-monospaced">
      Text <br />
      <a href="#">Link</a>
    </div>
  </Poster>
);

knobs.story = { parameters: { docs: { disable: true } } };
