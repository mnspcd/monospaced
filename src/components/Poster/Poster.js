import PropTypes from "prop-types";
import React from "react";

import "./Poster.css";

const Poster = ({ children, padding }) => (
  <div className="Poster" style={{ padding }}>
    {children}
  </div>
);

Poster.propTypes = {
  children: PropTypes.node.isRequired,
  padding: PropTypes.string,
};

export default Poster;
