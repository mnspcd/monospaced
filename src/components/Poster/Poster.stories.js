import React from "react";
import { text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Poster from "./Poster";

import typography from "../../system/typography";

storiesOf("Components/Poster", module).add("Component", () => {
  return (
    <div style={{ fontFamily: `${typography["font-monospaced"]}` }}>
      <Poster padding={text("padding", "1.5em 1em")}>
        Text color <br />
        <a href="#">Link color</a>
      </Poster>
    </div>
  );
});
