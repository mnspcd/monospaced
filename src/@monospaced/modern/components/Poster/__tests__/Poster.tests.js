/* eslint-env jest */
import { render } from "enzyme";
import React from "react";

import Poster from "../Poster";

describe("Poster component", () => {
  it("should render correctly", () => {
    const component = render(<Poster />);
    expect(component).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    const component = render(<Poster>children</Poster>);
    expect(component).toMatchSnapshot();
  });

  it("should render padding correctly", () => {
    const component = render(<Poster padding="1em">children</Poster>);
    expect(component).toMatchSnapshot();
  });
});
