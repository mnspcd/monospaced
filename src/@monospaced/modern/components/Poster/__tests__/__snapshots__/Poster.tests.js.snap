// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster component should render children correctly 1`] = `
<div
  class="Poster"
>
  children
</div>
`;

exports[`Poster component should render correctly 1`] = `
<div
  class="Poster"
/>
`;

exports[`Poster component should render padding correctly 1`] = `
<div
  class="Poster"
  style="padding:1em"
>
  children
</div>
`;
